mono/mini/mini.c

```diff
@@ -1983,7 +1983,7 @@ mono_add_var_location (MonoCompile *cfg, MonoInst *var, gboolean is_reg, int reg
 static void
 mono_apply_volatile (MonoInst *inst, MonoBitSet *set, gsize index)
 {
-	inst->flags |= mono_bitset_test_safe (set, index) ? MONO_INST_VOLATILE : 0;
+	inst->flags |= mono_bitset_test_safe (set, index) ? MONO_INST_INDIRECT : 0;
```

Comment:
The function name mono_apply_volatile suggests we do volatile ? This change is not aligning with function name. Also, am getting ramped up on the code, but what barriers are we talking about here with the change ?

Reply:
1. I agree, and the variables too.
2. I'd have to double check, but uses of parameters/locals in the icall managed to native wrappers.

Reply:
The point was "don't optimize", not "flush to memory across threads".

Reply:
Ok, we should do the change correctly updating function names/vars too. Please update the PR.